File: convalida-validators/src/main/java/convalida/validators/OnlyNumberValidator.java
package convalida.validators;

import android.widget.EditText;
import java.util.regex.Pattern;

/**
 * @author Wellington Costa on 29/06/2017.
 */
public class OnlyNumberValidator extends AbstractValidator{

    private boolean required;

    public OnlyNumberValidator(
            EditText editText,
            String errorMessage,
            boolean autoDismiss,
            boolean required
    ) {
        super(editText, errorMessage, autoDismiss);
        this.required = required;
    }

    @Override public boolean isValid(String value) {
        value = value.replace(" ", "");
        return (!required || !value.isEmpty())
                && (value.isEmpty() || Pattern.compile("^\\d+$").matcher(value).matches());
    }

}

File: convalida-validators/src/main/java/convalida/validators/OnlyNumberValidator.kt
package convalida.validators

import android.widget.EditText
import java.util.regex.Pattern

/**
 * @author Wellington Costa on 29/06/2017.
 */
class OnlyNumberValidator(editText: EditText, errorMessage: String) : AbstractValidator(editText, errorMessage) {

    override fun isNotValid(value: String): Boolean {
        return !Pattern.compile("^\\d+$").matcher(value).matches()
    }

}
